feat: Migrating the trainer controller to python logger

seshapad opened the pull request:
<!-- Thank you for the contribution! -->

### Description of the change

This PR switches the logging module from tranformer logger to python logger.

### Related issue number

Closes https://github.com/foundation-model-stack/fms-hf-tuning/issues/194

### How to verify the PR

<!-- Please provide instruction or screenshots on how to verify the PR.-->

### Was the PR tested

<!-- Describe how PR was tested -->
- [x] I have added >=1 unit test(s) for every new method I have added.
- [ ] I have ensured all unit tests pass

ashokponkumar commented on tuning/trainercontroller/callback.py:
```diff
@@ -417,6 +417,7 @@ def on_init_end(
                             config_log_level_str = DEFAULT_TRIGGER_LOG_LEVEL
                     else:
                         control.config = DEFAULT_CONFIG
+                    logger.warning(f"LOG LEVELS: {log_levels}")
```
Why do we have a warning level comment here?

seshapad replied:
Addressed.

ashokponkumar commented on tuning/trainercontroller/operations/hfcontrols.py:
```diff
@@ -10,6 +10,8 @@
 # Local
 from .operation import Operation
 
+logger = logging.getLogger()
```
Is this the standard way logging is being initialized elsewhere? Is there no logger name given?

seshapad replied:
Added `__name__`. Addressed.

ashokponkumar reviewed (changes requested):
Few changes

kmehant reviewed (approved):
successfully transitions from transformers logger to python logging for trainer controller

Abhishek-TAMU commented:
Thank you for the PR. As a curiosity, I have one question here. 

`logger` defined [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/operations/hfcontrols.py#L13C1-L13C37) and [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/patience.py#L34), would have default log level of `warning`, and it its log level could be modified in 2 ways:

1- Setting root log level in a main file (OR in the file where these module are imported for usage) using `(logging.basicConfig(level=logging.DEBUG))` and then importing `hfcontrols/patience` module. 

2- Setting log level of `hfcontrols/patience` from main file using : `logging.getLogger('hfcontrols').setLevel(logging.DEBUG)` / `logging.getLogger('patience').setLevel(logging.DEBUG)`

Is the use case of these modules includes any of the above, to set log level of `hfcontrols/patience` module ?
Other than this changes look fine to me.

kmehant commented:
@seshapad FYA 🔝

seshapad commented:
> Thank you for the PR. As a curiosity, I have one question here.
> 
> `logger` defined [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/operations/hfcontrols.py#L13C1-L13C37) and [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/patience.py#L34), would have default log level of `warning`, and its log level could be modified in 2 ways:
> 
> 1- Setting root log level in a main file (OR in the file where these module are imported for usage) using `(logging.basicConfig(level=logging.DEBUG))` and then importing `hfcontrols/patience` module.
> 
> 2- Setting log level of `hfcontrols/patience` from main file using : `logging.getLogger('hfcontrols').setLevel(logging.DEBUG)` / `logging.getLogger('patience').setLevel(logging.DEBUG)`
> 
> Is the use case of these modules includes any of the above, to set log level of `hfcontrols/patience` module ? Other than this changes look fine to me.

Thank you @Abhishek-TAMU . Will look into these changes and get back to you.

seshapad commented:
> Thank you for the PR. As a curiosity, I have one question here.
> 
> `logger` defined [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/operations/hfcontrols.py#L13C1-L13C37) and [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/patience.py#L34), would have default log level of `warning`, and its log level could be modified in 2 ways:
> 
> 1- Setting root log level in a main file (OR in the file where these module are imported for usage) using `(logging.basicConfig(level=logging.DEBUG))` and then importing `hfcontrols/patience` module.
> 
> 2- Setting log level of `hfcontrols/patience` from main file using : `logging.getLogger('hfcontrols').setLevel(logging.DEBUG)` / `logging.getLogger('patience').setLevel(logging.DEBUG)`
> 
> Is the use case of these modules includes any of the above, to set log level of `hfcontrols/patience` module ? Other than this changes look fine to me.

@Abhishek-TAMU I have addressed the above comments. Request you to review.

Abhishek-TAMU commented:
> @Abhishek-TAMU I have addressed the above comments. Request you to review.

Thank you for the changes. I can see setting of log level for individual loggers in different modules in your changes. This looks good to me!
Feel free to move ahead, no further questions/comments from my side. Thanks!

anhuong commented on tuning/trainercontroller/callback.py:
```diff
@@ -45,7 +45,8 @@
 
-logger = logging.get_logger(__name__)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
```
Should this reuse the `DEFAULT_TRIGGER_LOG_LEVEL` env var?

anhuong commented on tuning/trainercontroller/operations/hfcontrols.py:
```diff
@@ -10,6 +10,9 @@
 from .operation import Operation
 
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
```
Why is the default log level for the trainer controller debug? In general it's warning to print minimal print messages and the user can increase the level as needed

ashokponkumar commented:
> Thank you for the PR. As a curiosity, I have one question here.
> 
> `logger` defined [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/operations/hfcontrols.py#L13C1-L13C37) and [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/54f0ac978b6018f3cc2dcb6bf8611281ddb9cea1/tuning/trainercontroller/patience.py#L34), would have default log level of `warning`, and its log level could be modified in 2 ways:
> 
> 1- Setting root log level in a main file (OR in the file where these module are imported for usage) using `(logging.basicConfig(level=logging.DEBUG))` and then importing `hfcontrols/patience` module.
> 
> 2- Setting log level of `hfcontrols/patience` from main file using : `logging.getLogger('hfcontrols').setLevel(logging.DEBUG)` / `logging.getLogger('patience').setLevel(logging.DEBUG)`
> 
> Is the use case of these modules includes any of the above, to set log level of `hfcontrols/patience` module ? Other than this changes look fine to me.

@Abhishek-TAMU There is no need to change the logging level within patience, isn't? It should just inherit whatever is set in logging module, isn't it?

Abhishek-TAMU commented:
> @Abhishek-TAMU There is no need to change the logging level within patience, isn't? It should just inherit whatever is set in logging module, isn't it?

@ashokponkumar Just to clarify, can I ask if you are mentioning the fact that, after the setting of root log level by `main` function of `SFTTrainer.py`, that root log level should be inherited by `patience.py` ?

ashokponkumar commented:
> > @Abhishek-TAMU There is no need to change the logging level within patience, isn't? It should just inherit whatever is set in logging module, isn't it?
> 
> @ashokponkumar Just to clarify, can I ask if you are mentioning the fact that, after the setting of root log level by `main` function of `SFTTrainer.py`, that root log level should be inherited by `patience.py` ?

Yes @Abhishek-TAMU . If we are setting the logging level for class/module logging, then every logger inherited from it will already have the level set, isn't it?

Abhishek-TAMU commented:
@ashokponkumar 

Yes, if **root** log level is set in any class/module using `logging.basicConfig(level=logging.DEBUG/WARNING/ERROR)` as I have mentioned [above](https://github.com/foundation-model-stack/fms-hf-tuning/pull/309#issuecomment-2299190335), and this code runs initially setting up the root log level of the repo, then subsequently any logger used in any of the other class/module in the same repo will **inherit log level from root logger**. 
Unless if you would want to specify the log level of logger of any module different than the root logger, then as mentioned above, `logger.setLevel()` function could be used, for that specific logger.

ashokponkumar commented:
> @ashokponkumar 
> 
> Yes, if **root** log level is set in any class/module using `logging.basicConfig(level=logging.DEBUG/WARNING/ERROR)` as I have mentioned [above](https://github.com/foundation-model-stack/fms-hf-tuning/pull/309#issuecomment-2299190335), and this code runs initially setting up the root log level of the repo, then subsequently any logger used in any of the other class/module in the same repo will **inherit log level from root logger**. 
> Unless if you would want to specify the log level of logger of any module different than the root logger, then as mentioned above, `logger.setLevel()` function could be used, for that specific logger.
> 
> 

@seshapad then we can remove the part where we are setting the logging levels.

seshapad commented:
> > @ashokponkumar
> > Yes, if **root** log level is set in any class/module using `logging.basicConfig(level=logging.DEBUG/WARNING/ERROR)` as I have mentioned [above](https://github.com/foundation-model-stack/fms-hf-tuning/pull/309#issuecomment-2299190335), and this code runs initially setting up the root log level of the repo, then subsequently any logger used in any of the other class/module in the same repo will **inherit log level from root logger**.
> > Unless if you would want to specify the log level of logger of any module different than the root logger, then as mentioned above, `logger.setLevel()` function could be used, for that specific logger.
> 
> @seshapad then we can remove the part where we are setting the logging levels.

This is done. Thank you @ashokponkumar @Abhishek-TAMU @anhuong

ashokponkumar commented:
@anhuong can we merge this PR now?

anhuong commented:
@Abhishek-TAMU please review this again when you get a chance, it looks good to me. If it looks good to you, you can approve and squash-merge it.

Abhishek-TAMU commented:
LGTM!!

Abhishek-TAMU reviewed (approved):
LGTM!!
